### ISSUE TITLE
Queries on RNA-seq and Isoseq data in braker

### ISSUE THREAD

**Issue by lidachuis** [Gaius-Augustus/BRAKER#804]
hi，
 I have a sample, and this sample  has rnaseq data and isoseq data

braker.pl --genome=genome.fasta --prot_seq pro.fasta --bam=rnaseq.bam --bam=iso_seq.bam --threads 32

Is this command running correctly?

I don't know much about how to use isoseq in braker3.

**Comment by KatharinaHoff** [comment]
No. The two bam files can currently not be processed by the same
container/in one braker.pl run. Unless you modified GeneMark-ETP locally
for this. Then, the command would still be wrong, —bam can only be provided
once.

lidachuis ***@***.***> schrieb am Mo. 15. Apr. 2024 um 03:55:

**Comment by lidachuis** [comment]
Hi
I found that there is a branch long_reads in braker3. Can this branch solve the situation of iso-seq and rna-seq data?
[https://github.com/Gaius-Augustus/BRAKER/blob/long_reads/docs/long_reads/long_read_protocol.md]

At the same time, I also tried docker braker3:isoseq, but there was an error in its operation. The error was that bamtools was not found.

**Comment by KatharinaHoff** [comment]
The missing bamtools issue is one that I really fail to understand. Someone else independently reported that a while ago, and I cannot reproduce it. Here is how it looks when I run the container in singularity:

```
singularity exec braker_lr.sif which braker.pl
/opt/BRAKER/scripts/braker.pl
singularity exec braker_lr.sif which bamtools
/usr/bin/bamtools
singularity exec braker_lr.sif echo $PATH
/opt/singularity-3.11.3/lib/:/opt/singularity-3.11.3/bin/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```
Is the PATH different on your system? bamtools sits in /usr/bin, and that should be in the PATH.

The long read protocol is outdated to some extent. You may find the TSEBRA combiner advice at the end helpful. The protocol itself still builds on cupcake, most people don't use cupcake, anymore. It also does not filter the isoseq transcripts very well, that should work a lot better with GeneMark-ETP.

**Comment by lidachuis** [comment]
Hi，KatharinaHoff
I failed to use docker's isoseq before, and now I have successfully executed isoseq using singularity.
Now I have a new question, that is, I generated a braker.gtf through isoseq of the container, and also generated a braker.gtf using the data of rnaseq. How can I integrate them?

**Comment by KatharinaHoff** [comment]
You can run TSEBRA to merge them.

lidachuis ***@***.***> schrieb am Do. 18. Apr. 2024 um 04:55:

**Comment by lidachuis** [comment]
Thank you. Then should I use TSEBRA in the long_reads branch or tesbra in the master branch?

**Comment by KatharinaHoff** [comment]
You probably want to test several runs, either way. Luckily TSEBRA is much
faster than BRAKER.

lidachuis ***@***.***> schrieb am Do. 18. Apr. 2024 um 05:35:

**Comment by cjfields** [comment]
@KatharinaHoff just checking, but is the original isoseq long read + Illumina short read + protein protocol [in the wiki](https://github.com/Gaius-Augustus/BRAKER/blob/master/docs/long_reads/long_read_protocol.md) still up to date? Based on the response above it sounds like you recommend other approaches.  If so, what would they be?

**Comment by KatharinaHoff** [comment]
Please use the container shown on the PAG poster from 2024 (
https://github.com/Gaius-Augustus/BRAKER/blob/master/docs/posters/poster_PAG2024.pdf)
to run BRAKER3 with isoseq data. Run long and short reads, separately. You
can merge the two prediction sets with TSEBRA.

**Comment by cjfields** [comment]
Awesome, thx @KatharinaHoff!

**Comment by DustinSokolowski** [comment]
Dear @KatharinaHoff,

Thank you for your insight. As you likely know, StringTie can perform a hybrid transcriptome assembly using long and short reads (PMID: 35648784), implemented with the `--mix` option in StringTie. My understanding of `braker3_lr.sif` is that it is traditional braker3, but with RNA hints generated from StringTie with the `--long` option.

In principle, I imagine braker3 would generate a gtf file if you manually provided RNA-seq hints from `StringTie --mix` rather than running braker3 once with "short", again with "long" and then integrating them with TSEBRA. Have you compared the output gtf file between an annotation using `StringTie --mix` to generate hints with short-read RNA-seq and ISO-seq vs. the TSEBRA method you described? Would you expect one method to outperform the other strongly?

**Comment by KatharinaHoff** [comment]
Stringtie is not the problem, it can perform hybrid assembly. Both GeneMark-ETP and BRAKER would need a new input argument for inputting the long reads bam file, though. And that is currently lacking. Since I am not a GeneMark-ETP developer, I am hesitant to do that.

You can easily "cheat" by changing the stringtie command in GeneMark-ETP with a hardcoded long read bam file. I have done this, it works. But it's a hard coded path, then.

**Comment by DustinSokolowski** [comment]
Thank you for the fast and thorough reply, this is very helpful.

Best!
Dustin
